sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/ProcessBundleHandler.java

```diff
@@ -663,10 +695,47 @@ private ImmutableMap<String, ByteString> monitoringData(BundleProcessor bundlePr
         ByteString payload = monitoringInfo.getPayload();
         String shortId =
             shortIds.getOrCreateShortId(monitoringInfo.toBuilder().clearPayload().build());
-        result.put(shortId, payload);
+        monitoringData.put(shortId, payload);
       }
     }
-    return result.build();
+    bundleProcessor
+        .getBundleProgressReporterAndRegistrar()
+        .updateIntermediateMonitoringData(monitoringData);
```

Comment:
I'm not finding where either IntermediateMonitoringData or FinalMonitoringData below is used. It looks like they are being newly set in this change. Are they unused? What am I missing?